tests/random_attack.py

```diff
@@ -573,7 +598,12 @@ def check_np_data(self):
 
     def resize_rows(self, nrows):
         curr_nrows = self.nrows
-        self.df.nrows = nrows
+        try:
+            self.df.nrows = nrows
+        except ValueError as e:
+            assert str(e) == "Cannot increase the number of rows in a keyed frame"
+            return
+
```

Comment:
Same here: you should know in advance whether resizing should or should not throw an error.
It should throw iff (a) frame.nkeys > 0, and (b) new_nrows > nrows.